airbyte-integrations/connectors/source-chargebee/source_chargebee/source.py

```diff
@@ -22,205 +22,63 @@
-
-class InvoiceStream(ChargebeeStream):
-    name = "invoice"
-    api = Invoice
-
-
-class OrderStream(ChargebeeStream):
-    name = "order"
-    api = Order
-
-
-class PlanStream(ChargebeeStream):
-    name = "plan"
-    api = Plan
-
-
-class AddonStream(ChargebeeStream):
-    name = "addon"
-    api = Addon
+from .streams import AddonStream, CustomerStream, InvoiceStream, OrderStream, PlanStream, SubscriptionStream
 
 
 class SourceChargebee(AbstractSource):
-    # Class variables
-    LIMIT = 100
-
     def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
         # Configure the Chargebee Python SDK
-        chargebee.configure(
-            api_key=config["site_api_token"],
-            site=config["site"],
-        )
+        chargebee.configure(api_key=config["site_api_key"], site=config["site"])
         try:
-            # Get one subscription to test connection
-            Subscription.list(
-                # Set limit
-                # to test on a small dataset
-                params={
-                    "limit": 1,
-                },
-            )
+            subscription_stream = SubscriptionStream(start_date=config["start_date"])
+            next(subscription_stream.read_records(sync_mode=SyncMode.full_refresh))
             return True, None
         except Exception as err:
-            # Should catch all exceptions
-            # which are already handled by
-            # Chargebee Python wrapper
+            # Should catch all exceptions which are already handled by Chargebee Python wrapper.
             # https://github.com/chargebee/chargebee-python/blob/5346d833781de78a9eedbf9d12502f52c617c2d2/chargebee/http_request.py
-            return False, str(err)
+            return False, repr(err)
 
     def streams(self, config) -> List[Stream]:
         # Configure the Chargebee Python SDK
-        chargebee.configure(
-            api_key=config["site_api_token"],
-            site=config["site"],
-        )
-        # Add the streams
-        streams = [
-            SubscriptionStream(),
-            CustomerStream(),
-            InvoiceStream(),
-            OrderStream(),
-            PlanStream(),
-            AddonStream(),
+        chargebee.configure(api_key=config["site_api_key"], site=config["site"])
+
+        kwargs = {"start_date": config["start_date"]}
+        product_catalog_version = 1
+        try:
+            plan_stream_records = PlanStream(**kwargs).read_records(sync_mode=SyncMode.full_refresh)
+            next(plan_stream_records)
+        except APIError as err:
+            if err.http_code == requests.codes.BAD_REQUEST and err.error_code == "api_restricted":
+                product_catalog_version = 2
+        except StopIteration:
+            # This means that user has access to `plan` stream, but there is no data.
+            pass
+
+        # Below streams are suitable for both `Product Catalog 1.0` and `Product Catalog 2.0`.
+        common_streams = [
```

Comment:
I think that this information must be added to the documentation so that people do not get confused and do not look for Plan and Addon streams, if we do not provide them to them in the `discover` method.

Reply:
Done.